redbot/core/config.py

```diff
@@ -648,3 +586,265 @@ def member(self, member: discord.Member) -> MemberGroup:
+        values which have not yet been set.
+
+        Returns
+        -------
+        types.coroutine
+            A coroutine object which must be awaited to yield a `dict` mapping
+            :code:`USER_ID -> data`.
+
+        """
+        return self._all_from_scope(self.USER)
+
+    def _all_members_from_guild(self, group: Group, guild_data: dict) -> dict:
+        ret = {}
+        for member_id, member_data in guild_data.items():
+            new_member_data = group.defaults
+            new_member_data.update(member_data)
+            ret[int(member_id)] = new_member_data
+        return ret
+
+    async def all_members(self, guild: discord.Guild=None) -> dict:
+        """Get data for all members.
+
+        If :code:`guild` is specified, only the data for the members of that
+        guild will be returned. As such, the dict will map
+        :code:`MEMBER_ID -> data`. Otherwise, the dict maps
+        :code:`GUILD_ID -> MEMBER_ID -> data`.
+
+        Note
+        ----
+        The return value of this method will include registered defaults for
+        values which have not yet been set.
+
+        Parameters
+        ----------
+        guild : `discord.Guild`, optional
+            The guild to get the member data from. Can be omitted if data
+            from every member of all guilds is desired.
+
+        Returns
+        -------
+        dict
+            A dictionary of all member data from the given scope.
+
+        """
+        ret = {}
+        if guild is None:
+            group = self._get_base_group(self.MEMBER)
+            dict_ = await group()
+            for guild_id, guild_data in dict_.items():
+                ret[int(guild_id)] = self._all_members_from_guild(
+                    group, guild_data)
+        else:
+            group = self._get_base_group(self.MEMBER, guild.id)
+            guild_data = await group()
+            ret = self._all_members_from_guild(group, guild_data)
+        return ret
+
+    async def _clear_scope(self, *scopes: str):
+        """Clear all data in a particular scope.
+        
+        The only situation where a second scope should be passed in is if
+        member data from a specific guild is being cleared.
+
+        If no scopes are passed, then all data is cleared from every scope.
+
+        Parameters
+        ----------
+        *scopes : str, optional
+            The scope of the data. Generally only one scope needs to be
+            provided, a second only necessary for clearing member data
+            of a specific guild.
+
+            **Leaving blank removes all data from this Config instance.**
+
+        """
+        if not scopes:
+            group = Group(identifiers=(self.unique_identifier),
+                          defaults={},
+                          spawner=self.spawner)
+        else:
+            group = self._get_base_group(*scopes)
+        await group.set({})
+
+    def clear_all(self):
```

Comment:
Why are these not coroutines?

Reply:
Just gonna more or less summarise our conversation from discord for the record: I made them functions which return a coroutine because it's a micro-optimisation which removes one extra `await` statement, so the task gets done slightly faster, however I agree with your sentiment that the clarity of the function (esp. in documentation) is more important than this micro-optimization, so I've agreed to make the changes.